src/CoreWCF.Primitives/src/CoreWCF/Dispatcher/StreamFormatter.cs

```diff
@@ -12,7 +12,7 @@
 
 namespace CoreWCF.Dispatcher
 {
-    internal class StreamFormatter
+    public class StreamFormatter
     {
```

Comment:
It looks like this class has been made public for 2 reasons. First, to expose the inner class MessageBodyStream. Can you promote that inner class to be it's own class and not a child as it seems like it's a generally useful class. The second reason is to make WrapperName and WrapperNamespace available to DataContractJsonSerializerOperationFormatter. I've followed the logic of this code and I believe we can keep StreamFormatter internal by pre-modifying the OperationDescription. In WebHttpBehavior.CreateDataContractJsonSerializerOperationFormatter an instance of DataContractJsonSerializerOperationFormatter is created, passing in an OperationDescription. I believe we can clone and modify the OD to have the same effect. Basically make a new OD from the original cloning the Messages instances. For Messages[0], get the MessagePartDescription using something similar to StreamFormatter.GetStreamPart and fix up messageDescription.Body.WrapperName and messageDescription.Body.WrapperNamespace the same way as is done in the DataContractJsonSerializerOperationFormatter constructor. Check if the OperationDescription.Messages array has a second entry and if it does, do the same thing there. This way you don't need to reference StreamFormatter in DataContractJsonSerializerOperationFormatter and it can stay internal.

Reply:
That seems to have worked.